### PR TITLE
Add experimental Logs proto

### PR DESCRIPTION
**Pull request by tigrannajaryan**
This is using the same ProtoBuf generation mechanism as OTLP traces and metrics but
keeps Logs proto message definition in this repo for now, during the experimentation
phase. Once we settle on a certain proto format for logs this will be proposed
as official extension of Logs in OTLP.

This change also generates in-memory structs for logs using pdatagen.

TODO:
Body needs to implement than `any` data type that Log Data Model requires.
It will be done in a future PR. I want to align AttributeKeyValue in OTLP
with `any` value first.

### CODE REVIEW

File: cmd/pdatagen/internal/log_structs.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+	element:    logRecord,
+}
+
+var logRecord = &messageStruct{
+	structName:     "LogRecord",
+	description:    "// LogRecord are experimental implementation of OpenTelemetry Log Data Model.\n",
+	originFullName: "logsproto.LogRecord",
+	fields: []baseField{
+		&primitiveTypedField{
+			fieldMame:       "Timestamp",
```
Comment (keitwb) [inline cmd/pdatagen/internal/log_structs.go:67]: Is this supposed to be `fieldName`?

Comment (tigrannajaryan) [reply cmd/pdatagen/internal/log_structs.go:67]: Yes. Fixed.

**Comment by nilebox** [comment]
> keeps Logs proto message definition in this repo for now, during the experimentation
phase

You may consider putting it under `v1alpha1` version in the proto repository. Following the Kubernetes API convention, alpha versions don't provide any guarantees on maintaining and backward compatibility: https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning

**Comment by tigrannajaryan** [comment]
>You may consider putting it under v1alpha1 version in the proto repository.

Working on proto repo is slow since it requires multiple approvals from people who do not work on the Collector. It is too slow for experimentation and planned proof-of-concept that needs this proto.

**Pull request merged**